File: core/src/main/java/dev/bee/kanjianki/core/SettingValuePolicy.java
package dev.bee.kanjianki.core;

public final class SettingValuePolicy {
    private SettingValuePolicy() {
    }

    public static int parseInt(String value, int fallback) {
        try {
            return Integer.parseInt(value);
        } catch (NumberFormatException error) {
            return fallback;
        }
    }

    public static long parseLong(String value, long fallback) {
        try {
            return Long.parseLong(value);
        } catch (NumberFormatException error) {
            return fallback;
        }
    }

    public static double parseDouble(String value, double fallback) {
        try {
            return Double.parseDouble(value);
        } catch (NumberFormatException error) {
            return fallback;
        }
    }
}

File: core/src/main/kotlin/dev/bee/kanjianki/core/SettingValuePolicy.kt
package dev.bee.kanjianki.core

object SettingValuePolicy {
    @JvmStatic
    fun parseInt(value: String?, fallback: Int): Int {
        return value?.toIntOrNull() ?: fallback
    }

    @JvmStatic
    fun parseLong(value: String?, fallback: Long): Long {
        return value?.toLongOrNull() ?: fallback
    }

    @JvmStatic
    fun parseDouble(value: String?, fallback: Double): Double {
        return value?.toDoubleOrNull()?.takeIf { it.isFinite() } ?: fallback
    }
}
